railties/lib/rails/generators/testing/assertions.rb

```diff
@@ -23,7 +23,7 @@ module Assertions
         #     end
         #   end
         def assert_file(relative, *contents)
-          absolute = File.expand_path(relative, destination_root).shellescape
+          absolute = File.expand_path(relative, destination_root).gsub(" ", "\\ ")
```

Comment:
Can you point me in the direction why this `gsub` is needed?

Given:

```
$ ls spaces
file with spaces.txt v0.1.4~alpha+nightly
```

The following script outputs:

```ruby
path = File.expand_path("../spaces/file with spaces.txt", __FILE__)
path # => /Users/senny/Projects/ruby/spaces/file with spaces.txt
File.exist?(path) # => true

path = File.expand_path("../spaces/v0.1.4~alpha+nightly", __FILE__)
path # => /Users/senny/Projects/ruby/spaces/v0.1.4~alpha+nightly
File.exist?(path) # => true
```